### DISCUSSION TITLE
Is here some way to let ccls server support C++20

### DISCUSSION THREAD

**Discussion by PeterlitsZo** [neoclide/coc.nvim#2784] [Q&A] [answered]
I find someway in Google but not get useful answer. Thanks a lot. lol.

I want to just edit the `:CocConfig` if it can. Here is my `:CocConfig`:
```
{
    "languageserver": {
        "ccls": {
            "command": "ccls",
            "filetypes": [
                "c",
                "cpp",
                "objc",
                "objcpp"
            ],
            "rootPatterns": [
                ".ccls",
                "compile_commands.json",
                ".vim/",
                ".git/",
                ".hg/"
            ],
            "initializationOptions": {
                "cache": {
                    "directory": "/tmp/ccls"
                }
            },
            "clang.extraArgs": ["-std=c++20"]
        }
    },
    "prettier": {
        "tabWidth": 4
    }
}
```

But it do not work for me...

**Comment by kvreinzapp** [comment, marked as answer]
add `.ccls` file in the root path of my project works for me.
add these lines in `.ccls` file:
```
clang++
%h %cpp -std=c++20
```

(https://github.com/MaskRay/ccls/wiki/Project-Setup#ccls-file)
(https://github.com/MaskRay/ccls/issues/802)
(https://codevion.github.io/index.html#!vim/coc.md)

**Comment by fannheyward** [comment]
> "clang.extraArgs": ["-std=c++20"]

You're using `ccls`, but set args for `clangd`, they're not same thing.
